pkg/splunk/enterprise/util.go

```diff
@@ -307,19 +322,22 @@ func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.Me
 		mapAppListing["app-list-cluster.yaml"] = clusterAppsConf
 	}
 
+	if clusterAppsWithPreConf != yamlClusterAppsWithPreConfHeader {
+		mapAppListing["app-list-cluster-with-pre-config.yaml"] = clusterAppsWithPreConf
+	}
 	// Create App list config map
 	configMapName := GetSplunkAppsConfigMapName(cr.GetName(), crKind)
 	appListingConfigMap := splctrl.PrepareConfigMap(configMapName, cr.GetNamespace(), mapAppListing)
 
 	appListingConfigMap.SetOwnerReferences(append(appListingConfigMap.GetOwnerReferences(), splcommon.AsOwner(cr, true)))
 
-	if len(appListingConfigMap.Data) > 0 {
-		configMapDataChanged, err = splctrl.ApplyConfigMap(client, appListingConfigMap)
+	//if len(appListingConfigMap.Data) > 0 {
```

Comment:
So we are applying the updated ConfigMap whether its empty or not, correct?  Will this cause an extra restart if the list is emptied out?

Reply:
I'm thinking if we have a reconcile with app list changes, followed by a reconcile without any app changes.

Reply:
Good point Jeff. I removed this check as part of the reset logic, let me restore this